### PR TITLE
Integrate Hook multi plexer and MultiFactor validator

### PR DESCRIPTION
**Pull request by YasseinBilal**

### CODE REVIEW

File: src/module/hook-multi-plexer/types.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Address, Hex } from 'viem'
+
+export type SigHookInit = {
+  sig: Hex
+  subHooks: Address[]
+}
+
+export enum HookType {
```
Comment (kopy-kat) [inline src/module/hook-multi-plexer/types.ts:8]: in typescript, do these also correspond to indexes or are they strings?

File: src/module/hook-multi-plexer/usage.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    }
+  } catch {
+    throw new Error(`Failed to add hook ${hook}`)
+  }
+}
+
+export const addSigHook = ({
+  hook,
+  sig,
+  hookType,
```
Comment (kopy-kat) [inline src/module/hook-multi-plexer/usage.ts:55]: could it make sense to collapse these into a single func with an optional sig param? this is annoying in solidity hence isnt the case in the contract but could work here right?

File: src/module/hook-multi-plexer/types.ts (hunk 1 above)
Comment (YasseinBilal) [reply src/module/hook-multi-plexer/types.ts:8]: @kopy-kat zero based indexes unless you pass explicit values

File: src/module/hook-multi-plexer/usage.ts (hunk 2 above)
Comment (YasseinBilal) [reply src/module/hook-multi-plexer/usage.ts:55]: @kopy-kat Yeah make sense, will do with the remove hook as well

**Pull request merged**